### DISCUSSION TITLE
Support HTTPS connections to proxy servers from self-hosted runners

### DISCUSSION THREAD

**Discussion by hvpareja** [community/community#54345] [Actions]
### Select Topic Area

Product Feedback

### Body

Hello there,

I would like to raise a feature request for the GitHub Actions Runners.

Currently, we can make use of proxy servers to tunnel the runners' connections to Github, but the runner only supports the HTTP protocol in the connection to the proxy. See: https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/using-a-proxy-server-with-self-hosted-runners

Unfortunately, our enterprise proxy only allows HTTPS connections from its clients, but we found it is not possible from the runner.

Therefore, I would like to propose the creation of a new feature issue to enable HTTPS support for proxy connections.

**Comment by tenjaa** [comment]
Are you talking about the connection from the client to the proxy?
> the runner only supports the HTTP protocol in the connection to the proxy

Or from the proxy to the internet?
> our enterprise proxy only allows HTTPS connections from its clients

Because the second requirement to only allow https traffic sounds more like an enterprise requirement.
But you should be able to connect to the proxy via http and then afterwards to some https source on the internet.

**Reply by hvpareja** [reply]
Thanks @tenjaa for your answer!

You are correct about everything. For these reason, the runner does not support HTTPS protocol for proxy connections, I would like to raise this discussion as a feature request for the runner.

How can we make such suggestion to the developers?
